### ISSUE TITLE
TaskDefinition must be blank when using external Deployment controller

### ISSUE THREAD

**Issue by ttyz** [aws/aws-cdk#6245]
When using `DeploymentControllerType.EXTERNAL`, that was implemented in  #5402, CloudFormation returns an error

### Reproduction Steps
This is the code that I use:
```
        const service = new ecs.Ec2Service(this, "Service", {
            cluster,
            taskDefinition,
            deploymentController: {
                type: ecs.DeploymentControllerType.EXTERNAL
            }
        })
```

### Error Log

```
UPDATE_FAILED | AWS::ECS::Service  |  TaskDefinition must be blank.
```

### Environment

  - **CLI Version      : 1.23.0 (build 01f326e)**

### Other

This is my workaround for now:
```
        const cfnService = service.node.findChild('Service') as ecs.CfnService
        cfnService.addPropertyOverride("TaskDefinition", "")
        cfnService.addPropertyOverride("LaunchType", "")
```

--- 

This is :bug: Bug Report

**Comment by efekarakus** [comment]
Hi @ttyz, if I'm understanding the issue correctly would you expect the `taskDefinition` to not be a required prop if a `ecs.DeploymentControllerType.EXTERNAL` is set?

**Comment by ttyz** [comment]
Hi @efekarakus, Yes, I guess this would be the best solution, otherwise `ecs.DeploymentControllerType.EXTERNAL` does not seem to be useful, as the actual deployment will always fail. Also a warning that the `taskDefinition` will not be used in this case (when one was provided), is potentially helpful,

Other possible solution would be to clear the `taskDefinition` during synthesis, but this would require  some `taskDefinition` to be set, which is not very intuitive

**Comment by SoManyHs** [comment]
Should be addressed by #7130 . Thanks @ttyz for bringing this to our attention!
